lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -41,37 +41,37 @@
 /**
  * Helper code that wraps the YARN Client API for our purposes.
  */
-public final class YarnSubmissionHelper implements Closeable{
+public final class YarnSubmissionHelper implements Closeable {
+
   private static final Logger LOG = Logger.getLogger(YarnSubmissionHelper.class.getName());
 
   private final YarnClient yarnClient;
-  private final YarnClientApplication yarnClientApplication;
   private final GetNewApplicationResponse applicationResponse;
   private final ApplicationSubmissionContext applicationSubmissionContext;
   private final ApplicationId applicationId;
   private final Map<String, LocalResource> resources = new HashMap<>();
-  private final REEFFileNames fileNames;
   private final ClasspathProvider classpath;
   private final SecurityTokenProvider tokenProvider;
   private final List<String> commandPrefixList;
+
   private String driverStdoutFilePath;
   private String driverStderrFilePath;
-  private Class launcherClazz;
   private List<String> configurationFilePaths;
+  private Class launcherClass = REEFLauncher.class;
 
   public YarnSubmissionHelper(final YarnConfiguration yarnConfiguration,
                               final REEFFileNames fileNames,
                               final ClasspathProvider classpath,
                               final SecurityTokenProvider tokenProvider,
                               final List<String> commandPrefixList) throws IOException, YarnException {
-    this.fileNames = fileNames;
+
     this.classpath = classpath;
 
     this.driverStdoutFilePath =
-        ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/" + this.fileNames.getDriverStdoutFileName();
+        ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/" + fileNames.getDriverStdoutFileName();
 
     this.driverStderrFilePath =
-        ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/" + this.fileNames.getDriverStderrFileName();
+        ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/" + fileNames.getDriverStderrFileName();
```

Comment:
same here. I've got rid of `fileNames` as a member variable - now it's a constructor parameter only